static/js/copy_and_paste.js

```diff
@@ -165,64 +266,68 @@ function copy_handler() {
         }
     }
 
+    return {
+        ranges: ranges,
+        start_id: start_id,
+        end_id: end_id,
+        skip_same_td_check: skip_same_td_check,
+    };
+};
+
+// Copies the complete message(s) irrespective of how much text is selected.
+// Exact same code as copy_handler but here we handle single and multi message selections alike.
+exports.copy_full_message = function () {
+    var selection = window.getSelection();
+    var analysis = exports.analyze_selection(selection);
+    var ranges = analysis.ranges;
+    var start_id = analysis.start_id;
+    var end_id = analysis.end_id;
+    var div = $('<div>');
+
+    construct_copy_div(div, start_id, end_id);
+
+    select_div(div, selection);
+    document.execCommand('copy');
+    remove_div(div, ranges);
+};
+
+// Copies only the selected text irrespective of how many messages are selected.
+exports.copy_selected_text = function () {
+    var selection = window.getSelection();
+    var analysis = exports.analyze_selection(selection);
+    var ranges = analysis.ranges;
+    var start_id = analysis.start_id;
+    var end_id = analysis.end_id;
+    var skip_same_td_check = analysis.skip_same_td_check;
+    var div = $('<div>');
+    var selected_text = selection.toString().trim();
+
+    // Check if single message is selected, then let browser handle it.
     if (start_id === undefined || end_id === undefined) {
-        // In this case either the starting message or the ending
-        // message is not defined, so this is definitely not a
-        // multi-message selection and we can let the browser handle
-        // the copy.
+        document.execCommand('copy');
         return;
     }
 
     if (!skip_same_td_check && start_id === end_id) {
-        // Check whether the selection both starts and ends in the
-        // same message.  If so, Let the browser handle this.
+        document.execCommand('copy');
         return;
     }
 
-    // We've now decided to handle the copy event ourselves.
-    //
-    // We construct a temporary div for what we want the copy to pick up.
-    // We construct the div only once, rather than for each range as we can
-    // determine the starting and ending point with more confidence for the
-    // whole selection. When constructing for each `Range`, there is a high
-    // chance for overlaps between same message ids, avoiding which is much
-    // more difficult since we can get a range (start_id and end_id) for
-    // each selection `Range`.
-    construct_copy_div(div, start_id, end_id);
+    // Multiple messages selected, so we handle it manually.
+    alert("with trim: \n" + selected_text.trim());
```

Comment:
It seems you forgot to remove this debugging code :)

Reply:
Was just about to force push a commit to remove it. Thank you for pointing that out. :)